src/economy/economy.cpp

```diff
@@ -95,24 +653,35 @@ int32_t previous_gdp_record_index(sys::state& state) {
 	return ((date.year * 4 + date.month / 3) + gdp_history_length - 1) % gdp_history_length;
 }
 
-float ideal_pound_conversion_rate(sys::state& state, dcon::nation_id n) {
-	return state.world.nation_get_life_needs_costs(n, state.culture_definitions.primary_factory_worker)
-		+ 0.1f * state.world.nation_get_everyday_needs_costs(n, state.culture_definitions.primary_factory_worker);
+float ideal_pound_conversion_rate(sys::state& state, dcon::market_id n) {
+	return state.world.market_get_life_needs_costs(n, state.culture_definitions.primary_factory_worker)
+		+ 0.1f * state.world.market_get_everyday_needs_costs(n, state.culture_definitions.primary_factory_worker);
 }
 
-float gdp_adjusted(sys::state& state, dcon::nation_id n) {
-	float raw = state.world.nation_get_gdp(n);
+float gdp_adjusted(sys::state& state, dcon::market_id n) {
+	float raw = state.world.market_get_gdp(n);
 	float ideal_pound = ideal_pound_conversion_rate(state, n);
 	return raw / ideal_pound;
 }
 
+float gdp_adjusted(sys::state& state, dcon::nation_id n) {
+	auto total = 0.f;
+	state.world.nation_for_each_state_ownership(n, [&](auto soid) {
+		auto sid = state.world.state_ownership_get_state(soid);
+		auto market = state.world.state_instance_get_market_from_local_market(sid);
+		total = total + economy::gdp_adjusted(state, market);
+	});
+	return total;
+}
+
 float full_spending_cost(sys::state& state, dcon::nation_id n);
 void populate_army_consumption(sys::state& state);
 void populate_navy_consumption(sys::state& state);
 void populate_construction_consumption(sys::state& state);
 
 float commodity_daily_production_amount(sys::state& state, dcon::commodity_id c) {
-	return state.world.commodity_get_total_production(c);
+	// TODO
```

Comment:
Does this need to be filled for any critical bits of the ui?

Reply:
It's not used at all.